views/js/test/qtiCommonRenderer/interactions/choice/test.js

```diff
@@ -396,6 +397,31 @@ define([
         .render($container);
     });
 
+    QUnit.asyncTest('check dashes and dots in the identifier', function(assert){
+        var $container = $('#' + outsideContainerId);
+
+        QUnit.expect(4);
+
+        runner = qtiItemRunner('qti', badIdentChoiceData)
+            .on('render', function(){
+                var $discovery = $('.qti-choice[data-identifier="Discovery-new.dot"]', $container);
+                assert.equal($discovery.length, 1, 'the Discovery-new.dot choice exists');
+
+                $discovery.trigger('click');
+                this.setState({ RESPONSE : { response : {  base : { identifier : 'Discovery-new.dot' } } } });
```

Comment:
Why do you set the state manually ? Clicking the choice should change the state and the response and trigger the event. Could you please remove this line ?

Reply:
fixed